File: _submissions/18UEI056_Prashun/Day1/Sort_Binary_LL.cpp
/*Given a Linked List A consisting of N nodes.

The Linked List is binary i.e data values in the linked list nodes consist of only 0's and 1's.

You need to sort the linked list and return the new linked list.
*/
//code
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
ListNode* Solution::solve(ListNode* A) {
    ListNode* temp=A;
    ListNode* curr=A;
    int count1=0,count0=0;
    while(curr!=NULL)
    {if(curr->val==0)
    count0++;
    else
    {count1++;}
        curr=curr->next;
    }
    while(temp!=NULL)
    { if(count0 !=0)
    {
        temp->val=0;
        count0--;
    }
    else
    {temp->val=1;
    count1--;
    }
        temp=temp->next;
    }
    //A=curr;
    return A;
    
}

File: _submissions/19UCS159_Tharun/Day1/sortBinaryLL.java
/**
 * Definition for singly-linked list. class ListNode { public int val; public
 * ListNode next; ListNode(int x) { val = x; next = null; } }
 */
public class Solution {
  public ListNode solve(ListNode A) {
    int zeroes = 0;
    ListNode curr = A;
    while (curr != null) {
      if (curr.val == 0)
        zeroes++;
      curr = curr.next;
    }
    curr = A;
    while (curr != null) {
      if (zeroes > 0) {
        curr.val = 0;
        zeroes--;
      } else {
        curr.val = 1;
      }
      curr = curr.next;
    }
    return A;
  }
}
